### PR TITLE
[SystemZ][z/OS] This change adds support for the PPA2 section in zOS

### PR DESCRIPTION
**Pull request by ysyeda**
This PR adds support for the PPA2 fields.

### CODE REVIEW

**Review by uweigand — APPROVED** [review]
LGTM

**Comment by uweigand** [comment]
SystemZ back-end changes LGTM.    Additional reviews as to the Clang and time support changes would be appreciated.

**Comment by ysyeda** [comment]
@labath can you please review the changes in chrono.h/.cpp?
@MaskRay can you please review the changes in clang?

File: clang/test/CodeGen/SystemZ/systemz-ppa2.c [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+//   + the label for the PPA2 seems to vary on different versions.
+//     We try to cover all cases, and use substitution blocks to
+//     help write the tests. The contents of the PPA2 itself should
+//     not be different.
+//   + the [[:space:]] combines the two .byte lines into one pattern.
+//     This is necessary because if the lines were separated, the first
+//     .byte (i.e., the one for the 3) would, it seems, also match
+//     the .byte line below for the 34.
+
+// RUN: %clang --target=s390x-ibm-zos -xc -S -o - %s | FileCheck %s --check-prefix CHECK-C
```
Comment (MaskRay) [inline clang/test/CodeGen/SystemZ/systemz-ppa2.c:13]: Use `%clang_cc1` in non-driver tests

File: llvm/include/llvm/Support/Chrono.h [hunk 2]
```diff
@@ -33,13 +33,33 @@ namespace sys {
 template <typename D = std::chrono::nanoseconds>
 using TimePoint = std::chrono::time_point<std::chrono::system_clock, D>;
 
+// utc_clock and utc_time are only available since C++20. Add enough code to
+// support formatting date/time in UTC.
+class UtcClock : public std::chrono::system_clock {};
```
Comment (MaskRay) [inline llvm/include/llvm/Support/Chrono.h:38]: This Support change needs to be split from this patch with a unittest in llvm/unittest/Support.

Many people will think this z/OS patch is unrelated to them.

**Review by MaskRay — CHANGES_REQUESTED** [review]

File: clang/include/clang/Basic/LangStandard.h [hunk 3]
```diff
@@ -43,6 +43,7 @@ enum class Language : uint8_t {
   HLSL,
   ///@}
 };
+const char *LanguageToString(Language L);
```
Comment (MaskRay) [inline clang/include/clang/Basic/LangStandard.h:46]: `languageToString`? This file correctly uses `functionName` unlike many other Clang files...

File: llvm/include/llvm/Support/Chrono.h (hunk 2 above)
Comment (ysyeda) [reply llvm/include/llvm/Support/Chrono.h:38]: I created a new PR for the UTC clock changes here: https://github.com/llvm/llvm-project/pull/67846

**Comment by ysyeda** [comment]
Breaking this PR into 2 for chrono.h/.cpp changes and for zOS specific changes. 
The PR for UTC clock support is here: https://github.com/llvm/llvm-project/pull/65407

**Pull request closed without merging**